File: dotnet/src/Extensions/TemplateEngine.Basic/Blocks/Symbols.cs
// Copyright (c) Microsoft. All rights reserved.

namespace Microsoft.SemanticKernel.TemplateEngine.Basic.Blocks;

internal static class Symbols
{
    internal const char BlockStarter = '{';
    internal const char BlockEnder = '}';

    internal const char VarPrefix = '$';
    internal const char NamedArgBlockSeparator = '=';

    internal const char DblQuote = '"';
    internal const char SglQuote = '\'';
    internal const char EscapeChar = '\\';

    internal const char Space = ' ';
    internal const char Tab = '\t';
    internal const char NewLine = '\n';
    internal const char CarriageReturn = '\r';

    internal static bool IsVarPrefix(char c)
    {
        return (c == Symbols.VarPrefix);
    }

    internal static bool IsBlankSpace(char c)
    {
        return c is Symbols.Space or Symbols.NewLine or Symbols.CarriageReturn or Symbols.Tab;
    }

    internal static bool IsQuote(char c)
    {
        return c is Symbols.DblQuote or Symbols.SglQuote;
    }

    internal static bool CanBeEscaped(char c)
    {
        return c is Symbols.DblQuote or Symbols.SglQuote or Symbols.EscapeChar;
    }
}

File: python/semantic_kernel/template_engine/blocks/symbols.py
# Copyright (c) Microsoft. All rights reserved.
from enum import Enum


class Symbols(str, Enum):
    """Symbols used in the template engine."""

    BLOCK_STARTER = "{"
    BLOCK_ENDER = "}"

    VAR_PREFIX = "$"

    DBL_QUOTE = '"'
    SGL_QUOTE = "'"
    ESCAPE_CHAR = "\\"

    SPACE = " "
    TAB = "\t"
    NEW_LINE = "\n"
    CARRIAGE_RETURN = "\r"

    NAMED_ARG_BLOCK_SEPARATOR = "="
